contracts/interface/ITopicIdMapping.sol

```diff
@@ -1,41 +1,118 @@
 // SPDX-License-Identifier: GPL-3.0
 pragma solidity 0.8.27;
 
+/// @title ITopicIdMapping
+/// @notice Interface for a schema registry that maps topic IDs to structured metadata schemas
+/// @dev Each topic is uniquely identified and contains ABI-encoded arrays of field names and types
 interface ITopicIdMapping {
     /**
-     * @dev Definition of the structure of a Topic.
-     *
-     * Specification: Topics are information an issuer has about the identity holder.
-     * The structure should be as follows:
-     * format: A uint256 number which represents the format of the topic. The convention for what every format means will be shown in the documentation.
-     * name: A string which represents the name of the topic.
+     * @notice Struct that defines a registered topic schema
+     * @param name Human-readable name of the topic
+     * @param encodedFieldNames ABI-encoded string array representing field names
+     * @param encodedFieldTypes ABI-encoded string array representing field types (e.g. "uint256", "string[]")
      */
     struct Topic {
-        uint256 format;
         string name;
+        bytes encodedFieldNames;
```

Comment:
please rename the struct, it is confusing to call this Topic, while topic is a parameter of a claim.